snippets.go

```diff
@@ -157,8 +157,10 @@ func (s *SnippetsService) SnippetFileContent(snippet int, ref, filename string,
 // GitLab API docs:
 // https://docs.gitlab.com/ee/api/snippets.html#create-new-snippet
 type SnippetFile struct {
-	FilePath *string `url:"file_path,omitempty" json:"file_path,omitempty"`
-	Content  *string `url:"content,omitempty" json:"content,omitempty"`
+	FilePath     *string `url:"file_path,omitempty" json:"file_path,omitempty"`
+	Content      *string `url:"content,omitempty" json:"content,omitempty"`
+	Action       *string `json:"action,omitempty"`        // use only update
+	PreviousPath *string `json:"previous_path,omitempty"` // use only update
```

Comment:
Each method that needs different options should use its own options struct... So instead of adding the fields to this struct, maybe we should rename this struct to `CreateSnippetFile` and add an `UpdateSnippetFile` struct?

Reply:
I checked the parameters of the Update Snippet API and thought it could be handled by the same Strcut. 🤔
The reason is that we thought we could use the `SnippetFile` as it is, since the API json keys to be read as Struct are the same.
However, we were not able to consider the unity of the library as a whole.

> CreateSnippetFile and add an UpdateSnippetFile struct

I would like to implement this modification, if I may?